### DISCUSSION TITLE
Calling C#-assembly from vscode-extension

### DISCUSSION THREAD

**Discussion by hahoyer** [dotnet/interactive#1638] [Q&A]
I want to write a vscode-extension for my programming language - which is written in C#. When I searched for examples, I was told dotnet-interactive. But this is actually quite a huge project. 
Could you please help me to find the code part where I can see how to call a C#-assembly from TS/JS and how to assemble everything in a vscode-extension?

**Comment by jonsequitur** [comment]
You can create custom kernels and plug them into .NET Interactive by implementing your own `Kernel` class. Here's a very simple example in C#: https://github.com/dotnet/interactive/blob/600b134bce5e59802338bec3ef14ecd4268d92b6/src/Microsoft.DotNet.Interactive/HtmlKernel.cs#L11

A more complex example, with completions and other features, is here: https://github.com/dotnet/interactive/blob/f8e3d16b5e8325d43a5e1cf8951bfc5827fe0e52/src/Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs#L31

What kinds of functionality do you want in your VS Code extension? The .NET Interactive Notebooks extension will work with custom kernels if you want a notebook interface, in which case you wouldn't need to write any TypeScript or JavaScript to use the .NET Interactive kernel abstractions.

**Reply by hahoyer** [reply]
Syntax highlighting, IntelliSense, debugging, refactoring.
